packages/react/src/components/LocalNavigation/components/LocalNavigationDropdownItem/LocalNavigationDropdownItem.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { ReactNode, useCallback, useState } from 'react';
+import { AriaButtonProps } from '@react-aria/button';
+import { ChevronDown } from '@project44-manifest/react-icons';
+import { CSS, cx } from '@project44-manifest/react-styles';
+import type { ForwardRefComponent } from '@project44-manifest/react-types';
+import { Dropdown, DropdownProps } from '../../../Dropdown';
+import { LocalNavigationItem } from '../LocalNavigationItem';
+
+export interface LocalNavigationDropdownItemProps extends Omit<DropdownProps, 'children'> {
+  /** Should contain DropdownMenu with DropdownItems */
+  children: ReactNode;
+  /** Theme aware style object */
+  css?: CSS;
+  /** Props passed down to LocalNavigationItem */
+  localNavigationItemProps?: AriaButtonProps;
+  /** Title of LocalNavigationItem */
+  title: ReactNode | string;
+}
+
+export const LocalNavigationDropdownItem = React.forwardRef((props, forwardedRef) => {
+  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
+
+  const {
+    as,
+    children,
+    className: classNameProp,
+    css,
+    localNavigationItemProps,
+    title,
+    ...other
+  } = props;
+  const className = cx('manifest-local-navigation-dropdown-item', classNameProp);
+
+  const handleOpenChange = useCallback(
+    (isOpen: boolean) => {
+      setIsMenuOpen(isOpen);
+      other?.onOpenChange?.(isOpen);
+    },
+    [other],
+  );
+
+  return (
+    <Dropdown {...(other as DropdownProps)} onOpenChange={handleOpenChange}>
+      <LocalNavigationItem
+        {...localNavigationItemProps}
+        ref={forwardedRef}
+        as={as}
+        className={className}
+        css={{ gap: '$small', pr: '$small', typography: '$subtext-bold', ...css }}
+        isSelected={isMenuOpen}
+      >
+        {title}
+        <ChevronDown
+          css={{
+            transform: `rotate(${isMenuOpen ? -180 : 0}deg)`,
+            transformOrigin: '50% 54%',
+            transition: 'transform 100ms',
+          }}
+          size="small"
+        />
+      </LocalNavigationItem>
+      {children}
+    </Dropdown>
+  );
+}) as ForwardRefComponent<'button', LocalNavigationDropdownItemProps>;
```

Comment:
Why do we have to do a cast here?

Reply:
Unfortunately we don't have proper types set in `packages/react-types` for use in `forwardRef` types. We cannot use just a standard `HTMLButtonElement`, because it lack `as` and `className` props. The only types defined there are for returned forwarded component. This could be fixed in a separate PR, but for now I would stick with what is used across all components. All React components in Manifest are defined like that (except of few not using `as` and `className`).
![Screenshot 2023-08-10 at 10 33 57](https://github.com/project44/manifest/assets/90175251/1b3eff92-6f86-4b57-8b6f-87b23509dee0)